src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+                SearchSourceBuilder internalSearchSourceBuilder = new SearchSourceBuilder().query(boolQuery2).size(1).terminateAfter(1);
+                SearchRequest searchRequest = new SearchRequest(anomalyDetector.getIndices().get(0)).source(internalSearchSourceBuilder);
+                client.search(searchRequest, ActionListener.wrap(searchResponse -> {
+                    featureCounter.incrementAndGet();
+                    onFeatureAggregationValidation(searchResponse, feature, featureCounter);
+                }, exception -> onFailure(exception)));
+            }
+        }
+    }
+
+    private List<String> parseAggregationRequest(XContentParser parser) throws IOException {
+        List<String> fieldNames = new ArrayList<>();
+        XContentParser.Token token;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                final String field = parser.currentName();
+                switch (field) {
+                    case "field":
+                        parser.nextToken();
+                        fieldNames.add(parser.textOrNull());
+                        break;
+                    default:
+                        parser.skipChildren();
+                        break;
+                }
+            }
+        }
+        return fieldNames;
+    }
+
+    private void onFeatureAggregationValidation(SearchResponse response, Feature feature, AtomicInteger counter) throws IOException {
+        if (response.getHits().getTotalHits().value <= 0) {
+            String errorMsg = feature.getName() + ": feature query is potentially wrong as no hits were found";
+            suggestedChangesMap.computeIfAbsent("feature_attributes", k -> new ArrayList<>()).add(errorMsg);
+        }
+        if (counter.get() == anomalyDetector.getFeatureAttributes().size()) {
+            if (!suggestedChangesMap.isEmpty()) {
+                validateAnomalyDetectorResponse();
+            } else {
+                intervalValidation();
+            }
+        }
+    }
+
+    private long[][] createNewTimeRange(long detectorInterval) {
+        long timeRanges[][] = new long[MAX_NUM_OF_SAMPLES_VIEWED][2];
+        long delayMillis = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getWindowDelay())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
+        long dataEndTime = Instant.now().toEpochMilli() - delayMillis;
+        long dataStartTime = dataEndTime - ((long) (MAX_NUM_OF_SAMPLES_VIEWED) * detectorInterval);
+        for (int j = 0; j < MAX_NUM_OF_SAMPLES_VIEWED; j++) {
+            timeRanges[j][0] = dataStartTime + (j * detectorInterval);
+            timeRanges[j][1] = timeRanges[j][0] + detectorInterval;
+        }
+        return timeRanges;
+    }
+
+    private synchronized void intervalValidation() throws IOException {
```

Comment:
why making it `synchronized`? It looks it can only be executed when `counter == feature size`, aka when last feature response is returned.

Reply:
if `synchronized ` is not necessary, can we remove `synchronized` for `samplingHelper()`?

Reply:
I seemed to have issues running the detector interval inference when I took out the synchronized for this method, I did take it out for the samplingHelper which I changed names too verifyWithInterval. This is due to the fact that I had to use the notify and wait methods to handle the parallel execution of the elasticsearch queries since I am repeatedly calling queries in a loop while analyzing them.